Update

bbhtt opened the pull request:
See individual commits for details.

gtkmm and deps uses meson now (except gtkmm which doesn't support it at the current version it seems). This had to be done else build was failing due to issues like https://gitlab.gnome.org/GNOME/glibmm/-/issues/81

I'm going to make an issue to upstream about the patch, without that the build was failing due to `Wno-error=deprecated-declarations uint is deprecated`. But upstream isn't very active so let's see.  

I opened https://gitlab.com/utsushi/utsushi/-/issues/111

This built fine locally, needs testing though.

Also, updates Utsushi to the latest git commit which is from 1 year ago.

bbhtt committed:
Update submodule

bbhtt committed:
Add a gitignore

hfiguiere commented on gtkmm.json:
```diff
@@ -0,0 +1,142 @@
+{
```
please don't. It's harder to maintain.

bbhtt replied:
Ok, I can add it to the original manifest.

hfiguiere commented on com.github.utsushi.Utsushi.json:
```diff
@@ -34,69 +34,10 @@
 				"./b2 install"
             ]
 		},
-        "shared-modules/gudev/gudev.json",
+                "shared-modules/gudev/gudev.json",
 		"shared-modules/libusb/libusb.json",
```
indentation error. seems like your editor is misconfigured.

bbhtt replied:
Yea thanks.

hfiguiere commented on com.github.utsushi.Utsushi.json:
```diff
@@ -127,8 +68,12 @@
-					"commit": "ab7063358c9819dd37129c6ba58da1009302bdd7"
+					"commit": "839d06a5a80b353cb604eb9f7d352a1648ab1fdf"
 				},
+			        {
```
idnetation error too.

bbhtt commented:
The fonts on some of the buttons looks a bit weird, other than that I don't see visual issues.

Also somehow the screenshot is gone from GNOME Software, any ideas?

I'll remove the extra cleanups when I have a fix for the screenshot issue.

bbhtt commented:
> Also somehow the screenshot is gone from GNOME Software, any ideas?

Ok, this is probably https://github.com/flathub/flathub/issues/2434 because it appeared fine when I disabled flathub and installed from local repo.

> I'll remove the extra cleanups when I have a fix for the screenshot issue.

Done. This should be good to go. I switched the source url for boost because Sourceforge was giving download trouble and the new one seems to be ok and also what my distribution uses.

hfiguiere commented on com.github.utsushi.Utsushi.metainfo.xml:
```diff
@@ -25,9 +25,7 @@
-    <screenshot>
-      <image type="source">https://raw.githubusercontent.com/flathub/com.github.utsushi.Utsushi/master/screenshots/1.png</image>
-    </screenshot>
+      <screenshot><image width="786" height="616">https://raw.githubusercontent.com/flathub/com.github.utsushi.Utsushi/master/screenshots/1.png</image></screenshot>
```
not sure why you change this.

bbhtt replied:
I thought, image width and height are necessary of Flathub? Is it not?

I couldn't see the image, but it is probably the Flathub bug I linked above.

hfiguiere commented on com.github.utsushi.Utsushi.json:
```diff
@@ -127,7 +185,11 @@
 					"type": "git",
 					"url": "https://gitlab.com/utsushi/utsushi.git",
-					"commit": "ab7063358c9819dd37129c6ba58da1009302bdd7"
+					"commit": "839d06a5a80b353cb604eb9f7d352a1648ab1fdf"
```
any reason to change the revision?

bbhtt replied:
That's the latest commit, the manifest was following a commit  https://gitlab.com/utsushi/utsushi/-/commit/ab7063358c9819dd37129c6ba58da1009302bdd7 which is a year old and there were some translation updates and minor build related corrections.

Also a few minor fixes like https://gitlab.com/utsushi/utsushi/-/issues/103 were made during this time.

hfiguiere replied:
and CI build failures....

bbhtt replied:
Yea, I saw that but I haven't noticed those errors when building the Flatpak locally or here. My guess is the CI system change from Devuan Beowulf to Chimaera caused the issues, since it started failing with that commit. 

Anyway I can change it to the last commit that passed in the CI which is https://gitlab.com/utsushi/utsushi/-/commit/85d9379250db0cb3960d283a8886824680d4d314. Please let me know if you are fine with it.

hfiguiere commented on com.github.utsushi.Utsushi.json:
```diff
@@ -38,63 +42,117 @@
             "name": "atkmm",
             "sources": [
                 {
+                    "sha256": "a0bb49765ceccc293ab2c6735ba100431807d384ffa14c2ebd30e07993fd2fa4",
```
why change the order?

bbhtt replied:
The two order changes were unintentional, I'll fix it. Thanks.

hfiguiere commented on com.github.utsushi.Utsushi.json:
```diff
@@ -38,63 +42,117 @@
             "name": "pangomm",
             "sources": [
                 {
+                    "sha256": "57442ab4dc043877bfe3839915731ab2d693fc6634a71614422fb530c9eaa6f4",
```
and why change the order?

hfiguiere reviewed (changes requested):
it's nice to minmize change (like not reordering everything for no reason) for faster reviewing.

(some of the reorders not indicated, left to undo)

bbhtt committed:
Add patch to switch uintN to uintN_t to prevent build failure

bbhtt committed:
Update runtime and modules

- Update runtime to 42
- Update boost to 1.79.0 and switch to jfrog as source
- Add mm-common as build dep
- Update libsigc++ to 2.10.8
- Update glibmm to 2.66.3
- Update cairomm to 1.14.3
- Update pangomm to 2.46.2
- Update atkmm to 2.28.2
- Update gtkmm- to 2.24.5
- Switch everything to meson
- Update graphicsmagick to 1.3.38
- Add patch to manifest

hfiguiere commented:
bot, build
